modules/ROOT/pages/batch-processing-concept.adoc

```diff
@@ -192,32 +205,43 @@ During this phase, you can optionally configure the runtime to create a report o
+The fields in the report object (a `BatchJobResult`) are accessible as keys when using DataWeave selectors, such as `payload.failedRecords` to return the number of failed records in the instance.
 
-If you leave the On Complete phase empty and do not reference the batch job result object elsewhere in your application, the batch job simply completes, whether failed or successful.
+If you leave the On Complete phase empty and do not reference the batch job result object from elsewhere in your application, the batch job simply completes, whether failed or successful. However, after Mule completes execution of an entire batch job instance, the logs provide some processing information, for example:
 
-[TIP]
---
-As a good practice, you should configure some mechanism for reporting on failed or successful records to facilitate further action where required.
---
+----
+Finished execution for instance 'e84b5da0-fd5a-11ec-84a5-147ddaaf4f97'
+of job 'w-batch-take6Batch_Job'.
+Total Records processed: 1000. Successful records: 1000. Failed Records: 0
+----
 
+[[variable_propagation]]
 === Variable Propagation
 
-Every processed record of the Batch Job Instance starts with the same variables and values present before the execution of the block. Every record has its own set of variables, so new variables or modifications of already-existing variables during the processing of a given record will not be visible while processing another record. For each record, those variables (and modifications) are propagated through the different Batch Steps. For example, if record R1 sets a variable `varName: "hello"`, record R2 sets `varName: "world"`, and record R3 does not set this variable, then in the next step, R1 will see the value `"hello"`, R2 the value `"world"` and R3 will not see any value for that variable.
+Each record in a batch job instance inherits variables from the input to the Batch Job scope. You can access the values of these variables from within a Batch Step component but not within the Batch Aggregator scope.
 
-In the On Complete phase, none of these variables (not even the original ones) are visible. Only the final result is available in this phase. Moreover, since the Batch Job Instance executes asynchronously from the rest of the flow, no variable set in either a Batch Step or the On Complete phase will be visible outside the Batch Scope.
+Every record has its own set of variables, so new variables or modifications of already-existing variables during the processing of a given record will not be visible while processing another record. For each record, those variables (and modifications) are propagated through the different Batch Steps. For example, if record R1 sets a variable `varName: "hello"`, record R2 sets `varName: "world"`, and record R3 does not set this variable, then in the next step, R1 will see the value `"hello"`, R2 the value `"world"` and R3 will not see any value for that variable.
```

Comment:
IN PROGRESS: THIS HAS NOT BEEN INCORPORATED INTO THE NEW, PRECEDING PARAGRAPH, WHICH ALSO REQUIRES MORE WORK.